Code/RDGeneral/Dict.cpp

```diff
@@ -45,12 +51,27 @@ bool Dict::getValIfPresent(const std::string &what, std::string &res) const {
   //  other casts, which will then be lexically cast to type T.
   //
   for(size_t i=0; i< _data.size(); ++i) {
-    if (_data[i].key == what) {
+    if (_data[i].getKey() == tag) {
       rdvalue_tostring(_data[i].val, res);
       return true;
     }
   }
   return false;
 }
 
+
+namespace common_properties {
+
+const char *getPropName(int v) {
+  if (v>=0 && v<=MAX) {
+    return common_properties::propnames[v];
+  }
+  if ((size_t)v < RDKit::KeyIntPair::tagmap.keys.size())
+    return RDKit::KeyIntPair::tagmap.keys[v].c_str();
```

Comment:
Don't think we need to go as ugly as arrays of char. This will do the job of making sure the c_str() returned by getPropName are invariant for the lifetime of the process: 

```
std::vector<boost::shared_ptr<std::string> > keys;
```